MVVMIOS/ViewModel/MovieViewModel.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  MovieViewModel.swift
+//  MVVMIOS
+//
+//  Created by Ravi Bastola on 3/6/22.
+//
+
+import Foundation
+
+protocol MovieViewModelResultPresentable: AnyObject {
+    func presentResult(_ result: Result<[MovieResult], Error>?)
+}
+
+class MovieViewModel {
+
+    let service: NetworkingProtocol
+
+    weak var resultPresentable: MovieViewModelResultPresentable?
+
+    init(_ service: NetworkingProtocol) {
+        self.service = service
+    }
+
+    func fetchDataFromApi() {
+        guard let url = URL(string: "http://gdhistory.com/api/periods") else {
+            self.resultPresentable?.presentResult(.failure(NetworkError.invalidURL))
+            return
+        }
+
+        service.loadUsingModel([MovieResult].self, from: .init(url: url)) { result in
+            switch result {
+            case .success(let movieResult):
+                self.resultPresentable?.presentResult(.success(movieResult))
+            case .failure(let error as NetworkError):
+                switch error {
+                case .invalidURL:
+                    self.resultPresentable?.presentResult(.failure(NetworkError.invalidURL))
+                default:
+                    break
```

Comment:
Handle all cases, otherwise users will get stuck when this line does hit. The UI won't refresh but there will have been an error and they'll have no idea what to do. In this case you'd probably want to (following the other advice above) send something like `MovieViewState.error(.init(errorText: "Unexpected Error, please try again"))`. At least that way they know something happened and the best they can do is retry.